Tests failures when using tmux 3.7

With the latest and recent [release of tmux 3.7](https://github.com/tmux/tmux/releases/tag/3.7) ([CHANGES](https://raw.githubusercontent.com/tmux/tmux/3.7/CHANGES)) the following libtmux tests fail:

```
FAILED tests/test_pane.py::test_join_pane - libtmux.exc.LibTmuxException: break-pane: server exited unexpectedly
FAILED tests/test_pane.py::test_join_pane_horizontal - libtmux.exc.LibTmuxException: break-pane: server exited unexpectedly
FAILED tests/test_pane.py::test_break_pane_basic - libtmux.exc.LibTmuxException: break-pane: server exited unexpectedly
FAILED tests/test_window.py::test_window_rename[rename_with_escapes] - AssertionError: assert 'hello \\\\ wazzup 0' == 'hello \\ wazzup 0'
FAILED tests/test_pane.py::test_break_pane_with_name - AssertionError: assert 'coreutils' == 'my_broken'
```

<details><summary>Detailed test log output</summary>

```
=================================== FAILURES ===================================
________________________________ test_join_pane ________________________________
[gw1] darwin -- Python 3.13.13 /nix/store/mfkdmplffnbc0av8r6pknl796a6b1r2n-python3-3.13.13/bin/python3.13
tests/test_pane.py:1289: in test_join_pane
    new_window = new_pane.break_pane()
                 ^^^^^^^^^^^^^^^^^^^^^
        new_pane   = <[LibTmuxException('no server running on /private/tmp/tmux-351/libtmux_test6luv7qac') raised in repr()] Pane object at 0x10b62da90>
        pane       = <[LibTmuxException('no server running on /private/tmp/tmux-351/libtmux_test6luv7qac') raised in repr()] Pane object at 0x10b62d6d0>
        session    = Session($1 libtmux_fveud3ah)
        window     = <[LibTmuxException('no server running on /private/tmp/tmux-351/libtmux_test6luv7qac') raised in repr()] Window object at 0x10b62d590>
/nix/store/8mx489rggv1i2mkk9j8gxp7j16djclby-python3.13-libtmux-0.58.1/lib/python3.13/site-packages/libtmux/pane.py:2085: in break_pane
    raise_if_stderr(proc, "break-pane")
        detach     = True
        proc       = <libtmux.common.tmux_cmd object at 0x10b440130>
        self       = <[LibTmuxException('no server running on /private/tmp/tmux-351/libtmux_test6luv7qac') raised in repr()] Pane object at 0x10b62da90>
        tmux_args  = ('-P', '-F#{window_id}', '-d', '-s', '%3')
        window_name = None
/nix/store/8mx489rggv1i2mkk9j8gxp7j16djclby-python3.13-libtmux-0.58.1/lib/python3.13/site-packages/libtmux/common.py:277: in raise_if_stderr
    raise exc.LibTmuxException(
E   libtmux.exc.LibTmuxException: break-pane: server exited unexpectedly
        proc       = <libtmux.common.tmux_cmd object at 0x10b440130>
        subcommand = 'break-pane'
__________________________ test_join_pane_horizontal ___________________________
[gw1] darwin -- Python 3.13.13 /nix/store/mfkdmplffnbc0av8r6pknl796a6b1r2n-python3-3.13.13/bin/python3.13
tests/test_pane.py:1312: in test_join_pane_horizontal
    new_pane.break_pane()
        new_pane   = <[LibTmuxException('no server running on /private/tmp/tmux-351/libtmux_testav3xdoi4') raised in repr()] Pane object at 0x10b62e710>
        pane       = <[LibTmuxException('no server running on /private/tmp/tmux-351/libtmux_testav3xdoi4') raised in repr()] Pane object at 0x10b62e5d0>
        session    = Session($1 libtmux_s5wpu1i4)
        window     = <[LibTmuxException('no server running on /private/tmp/tmux-351/libtmux_testav3xdoi4') raised in repr()] Window object at 0x10b62d590>
/nix/store/8mx489rggv1i2mkk9j8gxp7j16djclby-python3.13-libtmux-0.58.1/lib/python3.13/site-packages/libtmux/pane.py:2085: in break_pane
    raise_if_stderr(proc, "break-pane")
        detach     = True
        proc       = <libtmux.common.tmux_cmd object at 0x10b440590>
        self       = <[LibTmuxException('no server running on /private/tmp/tmux-351/libtmux_testav3xdoi4') raised in repr()] Pane object at 0x10b62e710>
        tmux_args  = ('-P', '-F#{window_id}', '-d', '-s', '%3')
        window_name = None
/nix/store/8mx489rggv1i2mkk9j8gxp7j16djclby-python3.13-libtmux-0.58.1/lib/python3.13/site-packages/libtmux/common.py:277: in raise_if_stderr
    raise exc.LibTmuxException(
E   libtmux.exc.LibTmuxException: break-pane: server exited unexpectedly
        proc       = <libtmux.common.tmux_cmd object at 0x10b440590>
        subcommand = 'break-pane'
____________________________ test_break_pane_basic _____________________________
[gw1] darwin -- Python 3.13.13 /nix/store/mfkdmplffnbc0av8r6pknl796a6b1r2n-python3-3.13.13/bin/python3.13
tests/test_pane.py:1329: in test_break_pane_basic
    new_window = new_pane.break_pane()
                 ^^^^^^^^^^^^^^^^^^^^^
        initial_window_count = 2
        new_pane   = <[LibTmuxException('no server running on /private/tmp/tmux-351/libtmux_testj8qsmo7v') raised in repr()] Pane object at 0x10b932850>
        pane       = <[LibTmuxException('no server running on /private/tmp/tmux-351/libtmux_testj8qsmo7v') raised in repr()] Pane object at 0x10b931bd0>
        session    = Session($1 libtmux_f7csmko8)
        window     = <[LibTmuxException('no server running on /private/tmp/tmux-351/libtmux_testj8qsmo7v') raised in repr()] Window object at 0x10b931a90>
/nix/store/8mx489rggv1i2mkk9j8gxp7j16djclby-python3.13-libtmux-0.58.1/lib/python3.13/site-packages/libtmux/pane.py:2085: in break_pane
    raise_if_stderr(proc, "break-pane")
        detach     = True
        proc       = <libtmux.common.tmux_cmd object at 0x10b6afcb0>
        self       = <[LibTmuxException('no server running on /private/tmp/tmux-351/libtmux_testj8qsmo7v') raised in repr()] Pane object at 0x10b932850>
        tmux_args  = ('-P', '-F#{window_id}', '-d', '-s', '%3')
        window_name = None
/nix/store/8mx489rggv1i2mkk9j8gxp7j16djclby-python3.13-libtmux-0.58.1/lib/python3.13/site-packages/libtmux/common.py:277: in raise_if_stderr
    raise exc.LibTmuxException(
E   libtmux.exc.LibTmuxException: break-pane: server exited unexpectedly
        proc       = <libtmux.common.tmux_cmd object at 0x10b6afcb0>
        subcommand = 'break-pane'
___________________ test_window_rename[rename_with_escapes] ____________________
[gw3] darwin -- Python 3.13.13 /nix/store/mfkdmplffnbc0av8r6pknl796a6b1r2n-python3-3.13.13/bin/python3.13
tests/test_window.py:247: in test_window_rename
    assert window.window_name == window_name_before
E   AssertionError: assert 'hello \\\\ wazzup 0' == 'hello \\ wazzup 0'
E     
E     - hello \ wazzup 0
E     + hello \\ wazzup 0
E     ?        +
        session    = Session($1 libtmux_cpm104j8)
        test_id    = 'rename_with_escapes'
        window     = Window(@2 2:hello \\ wazzup 0, Session($1 libtmux_cpm104j8))
        window_name_after = 'hello \\\\ wazzup 0'
        window_name_before = 'hello \\ wazzup 0'
        window_name_input = 'hello \\ wazzup 0'
__________________________ test_break_pane_with_name ___________________________
[gw1] darwin -- Python 3.13.13 /nix/store/mfkdmplffnbc0av8r6pknl796a6b1r2n-python3-3.13.13/bin/python3.13
tests/test_pane.py:1346: in test_break_pane_with_name
    assert new_window.window_name == "my_broken"
E   AssertionError: assert 'coreutils' == 'my_broken'
E     
E     - my_broken
E     + coreutils
        new_pane   = Pane(%3 Window(@2 2:test_break_name, Session($1 libtmux_1x5rqmza)))
        new_window = Window(@3 3:coreutils, Session($1 libtmux_1x5rqmza))
        pane       = Pane(%2 Window(@2 2:test_break_name, Session($1 libtmux_1x5rqmza)))
        session    = Session($1 libtmux_1x5rqmza)
        window     = Window(@2 2:test_break_name, Session($1 libtmux_1x5rqmza))
=========================== short test summary info ============================
FAILED tests/test_pane.py::test_join_pane - libtmux.exc.LibTmuxException: break-pane: server exited unexpectedly
FAILED tests/test_pane.py::test_join_pane_horizontal - libtmux.exc.LibTmuxException: break-pane: server exited unexpectedly
FAILED tests/test_pane.py::test_break_pane_basic - libtmux.exc.LibTmuxException: break-pane: server exited unexpectedly
FAILED tests/test_window.py::test_window_rename[rename_with_escapes] - AssertionError: assert 'hello \\\\ wazzup 0' == 'hello \\ wazzup 0'
FAILED tests/test_pane.py::test_break_pane_with_name - AssertionError: assert 'coreutils' == 'my_broken'
=================== 5 failed, 877 passed, 11 rerun in 22.71s ===================
```

</details> 

What would be helpful for libtmux maintainers to get these resolved?

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -12,7 +12,7 @@ jobs:
       fail-fast: false
       matrix:
         python-version: ['3.14']
-        tmux-version: ['3.2a', '3.3a', '3.4', '3.5', '3.6', 'master']
+        tmux-version: ['3.2a', '3.3a', '3.4', '3.5', '3.6', '3.7', 'master']
     steps:
       - uses: actions/checkout@v6
 

--- a/CHANGES
+++ b/CHANGES
@@ -54,8 +54,8 @@ a NULL-pointer dereference whenever a pane is broken out of a window
 without an explicit name (fixed upstream after the 3.7 release).
 {meth}`~libtmux.Pane.break_pane` now works around this on tmux 3.7 by
 always supplying a name and applying the requested one afterwards, so
-breaking a pane into a new window — and {meth}`~libtmux.Pane.join`,
-which relies on it — work again. Older and post-3.7 tmux are
+breaking a pane into a new window — and joining it back with
+{meth}`~libtmux.Pane.join` — work again. Older and post-3.7 tmux are
 unaffected.
 
 ## libtmux 0.58.1 (2026-06-16)

--- a/CHANGES
+++ b/CHANGES
@@ -45,6 +45,19 @@ $ uvx --from 'libtmux' --prerelease allow python
 _Notes on the upcoming release will go here._
 <!-- END PLACEHOLDER - ADD NEW CHANGELOG ENTRIES BELOW THIS LINE -->
 
+### Fixes
+
+#### `break-pane` no longer crashes the tmux 3.7 server (#693)
+
+tmux 3.7 shipped a regression where `break-pane` aborts the server with
+a NULL-pointer dereference whenever a pane is broken out of a window
+without an explicit name (fixed upstream after the 3.7 release).
+{meth}`~libtmux.Pane.break_pane` now works around this on tmux 3.7 by
+always supplying a name and applying the requested one afterwards, so
+breaking a pane into a new window — and {meth}`~libtmux.Pane.join`,
+which relies on it — work again. Older and post-3.7 tmux are
+unaffected.
+
 ## libtmux 0.58.1 (2026-06-16)
 
 libtmux 0.58.1 restores compatibility with pytest 9.1. The bundled

--- a/docs/topics/filtering.md
+++ b/docs/topics/filtering.md
@@ -153,12 +153,12 @@ For complex patterns, use regex lookups:
 
 ```python
 >>> # Create windows with version-like names
->>> w1 = session.new_window(window_name="app-v1.0")
->>> w2 = session.new_window(window_name="app-v2.0")
+>>> w1 = session.new_window(window_name="app-v1-0")
+>>> w2 = session.new_window(window_name="app-v2-0")
 >>> w3 = session.new_window(window_name="app-beta")
 
 >>> # Match version pattern
->>> versioned = session.windows.filter(window_name__regex=r'v\d+\.\d+$')
+>>> versioned = session.windows.filter(window_name__regex=r'v\d+-\d+$')
 >>> len(versioned) >= 2
 True
 
@@ -238,19 +238,19 @@ True
 
 ```python
 >>> # Create windows following a naming convention
->>> w1 = session.new_window(window_name="project:frontend")
->>> w2 = session.new_window(window_name="project:backend")
+>>> w1 = session.new_window(window_name="project-frontend")
+>>> w2 = session.new_window(window_name="project-backend")
 >>> w3 = session.new_window(window_name="logs")
 
 >>> # Find all project windows
->>> project_windows = session.windows.filter(window_name__startswith='project:')
+>>> project_windows = session.windows.filter(window_name__startswith='project-')
 >>> len(project_windows) >= 2
 True
 
 >>> # Get specific project window
->>> backend = session.windows.get(window_name='project:backend')
+>>> backend = session.windows.get(window_name='project-backend')
 >>> backend.window_name
-'project:backend'
+'project-backend'
 
 >>> # Clean up
 >>> w1.kill()

--- a/docs/topics/traversal.md
+++ b/docs/topics/traversal.md
@@ -205,13 +205,13 @@ For complex patterns, use `__regex` or `__iregex`:
 
 ```python
 >>> # Create versioned windows
->>> w1 = session.new_window(window_name="release-v1.0")
->>> w2 = session.new_window(window_name="release-v2.0")
+>>> w1 = session.new_window(window_name="release-v1-0")
+>>> w2 = session.new_window(window_name="release-v2-0")
 >>> w3 = session.new_window(window_name="dev")
 
->>> # Match semantic version pattern
->>> session.windows.filter(window_name__regex=r'v\d+\.\d+')  # doctest: +ELLIPSIS
-[Window(@... ...:release-v1.0, Session($... ...)), Window(@... ...:release-v2.0, Session($... ...))]
+>>> # Match version pattern
+>>> session.windows.filter(window_name__regex=r'v\d+-\d+')  # doctest: +ELLIPSIS
+[Window(@... ...:release-v1-0, Session($... ...)), Window(@... ...:release-v2-0, Session($... ...))]
 
 >>> # Clean up
 >>> w1.kill()

--- a/src/libtmux/pane.py
+++ b/src/libtmux/pane.py
@@ -2069,11 +2069,10 @@ class Pane(
         >>> new_window.window_name
         'broken'
         """
-        # tmux 3.7 segfaults on break-pane when no -n is given (a NULL window
-        # name is dereferenced); fixed upstream after the 3.7 release. Pass a
-        # placeholder -n to avoid the crash -- 3.7 ignores it and applies the
-        # default name -- then rename to the requested name below. Gated on
-        # exactly 3.7 so fixed builds, which honor -n, are left untouched.
+        # tmux 3.7 segfaults break-pane when -n is absent and ignores -n when
+        # given (NULL-deref, fixed upstream after 3.7). Always pass -n -- the
+        # requested name or a placeholder -- then set the real name via
+        # rename-window below. Gated on exactly 3.7; other builds are unaffected.
         breaks_without_name = has_version("3.7", tmux_bin=self.server.tmux_bin)
 
         tmux_args: tuple[str, ...] = ("-P", "-F#{window_id}")
@@ -2100,7 +2099,6 @@ class Pane(
         window = Window.from_window_id(server=self.server, window_id=window_id)
 
         if window_name is not None and breaks_without_name:
-            # tmux 3.7 ignored the -n above; set the requested name now.
             window.rename_window(window_name)
 
         return window

--- a/src/libtmux/pane.py
+++ b/src/libtmux/pane.py
@@ -14,7 +14,7 @@ import typing as t
 import warnings
 
 from libtmux import exc
-from libtmux.common import has_gte_version, raise_if_stderr, tmux_cmd
+from libtmux.common import has_gte_version, has_version, raise_if_stderr, tmux_cmd
 from libtmux.constants import (
     PANE_DIRECTION_FLAG_MAP,
     RESIZE_ADJUSTMENT_DIRECTION_FLAG_MAP,
@@ -2069,6 +2069,13 @@ class Pane(
         >>> new_window.window_name
         'broken'
         """
+        # tmux 3.7 segfaults on break-pane when no -n is given (a NULL window
+        # name is dereferenced); fixed upstream after the 3.7 release. Pass a
+        # placeholder -n to avoid the crash -- 3.7 ignores it and applies the
+        # default name -- then rename to the requested name below. Gated on
+        # exactly 3.7 so fixed builds, which honor -n, are left untouched.
+        breaks_without_name = has_version("3.7", tmux_bin=self.server.tmux_bin)
+
         tmux_args: tuple[str, ...] = ("-P", "-F#{window_id}")
 
         if detach:
@@ -2076,6 +2083,8 @@ class Pane(
 
         if window_name is not None:
             tmux_args += ("-n", window_name)
+        elif breaks_without_name:
+            tmux_args += ("-n", "libtmux")
 
         tmux_args += ("-s", str(self.pane_id))
 
@@ -2088,7 +2097,13 @@ class Pane(
 
         from libtmux.window import Window
 
-        return Window.from_window_id(server=self.server, window_id=window_id)
+        window = Window.from_window_id(server=self.server, window_id=window_id)
+
+        if window_name is not None and breaks_without_name:
+            # tmux 3.7 ignored the -n above; set the requested name now.
+            window.rename_window(window_name)
+
+        return window
 
     def swap(
         self,

--- a/tests/test_window.py
+++ b/tests/test_window.py
@@ -219,8 +219,13 @@ WINDOW_RENAME_FIXTURES: list[WindowRenameFixture] = [
         window_name_after="ha ha ha fjewlkjflwef",
     ),
     WindowRenameFixture(
+        # Create with a plain name and only assert that rename doubles the
+        # backslash: window_set_name() has escaped names via vis(3) since tmux
+        # 2.6, so this holds on all supported versions. (new-window only began
+        # escaping names in tmux 3.7, so a backslash in the *create* name is
+        # version-dependent and must not be asserted here.)
         test_id="rename_with_escapes",
-        window_name_before=r"hello \ wazzup 0",
+        window_name_before="test",
         window_name_input=r"hello \ wazzup 0",
         window_name_after=r"hello \\ wazzup 0",
     ),

--- a/tests/test_window.py
+++ b/tests/test_window.py
@@ -257,6 +257,48 @@ def test_window_rename(
     assert window.window_name == window_name_after
 
 
+class WindowNameValidationFixture(t.NamedTuple):
+    """Test fixture for tmux 3.7 window-name validation."""
+
+    test_id: str
+    window_name: str
+
+
+# tmux 3.7 rejects ':' and '.' in window names server-side; tmux 3.2a-3.6
+# accept them. See https://github.com/tmux/tmux/issues/4999.
+WINDOW_NAME_INVALID_ON_3_7_FIXTURES: list[WindowNameValidationFixture] = [
+    WindowNameValidationFixture(test_id="colon", window_name="project:frontend"),
+    WindowNameValidationFixture(test_id="period", window_name="app-v1.0"),
+]
+
+
+@pytest.mark.parametrize(
+    list(WindowNameValidationFixture._fields),
+    WINDOW_NAME_INVALID_ON_3_7_FIXTURES,
+    ids=[tc.test_id for tc in WINDOW_NAME_INVALID_ON_3_7_FIXTURES],
+)
+def test_new_window_name_invalid_on_3_7(
+    session: Session,
+    test_id: str,
+    window_name: str,
+) -> None:
+    """Tmux 3.7+ rejects ':'/'.' in window names; older tmux accepts them.
+
+    The rejection is enforced by tmux itself (no client-side validation),
+    so it surfaces as a :class:`~libtmux.exc.LibTmuxException`. On tmux
+    3.2a-3.6 the same names are accepted verbatim.
+    """
+    from libtmux.common import has_gte_version
+
+    if has_gte_version("3.7"):
+        with pytest.raises(exc.LibTmuxException, match="invalid window name"):
+            session.new_window(window_name=window_name)
+    else:
+        window = session.new_window(window_name=window_name)
+        assert window.window_name == window_name
+        window.kill()
+
+
 def test_kill_window(session: Session) -> None:
     """Test window.kill() kills window."""
     session.new_window()

--- a/tests/test_window.py
+++ b/tests/test_window.py
@@ -264,7 +264,7 @@ class WindowNameValidationFixture(t.NamedTuple):
     window_name: str
 
 
-# tmux 3.7 rejects ':' and '.' in window names server-side; tmux 3.2a-3.6
+# tmux 3.7+ rejects ':' and '.' in window names server-side; tmux 3.2a-3.6
 # accept them. See https://github.com/tmux/tmux/issues/4999.
 WINDOW_NAME_INVALID_ON_3_7_FIXTURES: list[WindowNameValidationFixture] = [
     WindowNameValidationFixture(test_id="colon", window_name="project:frontend"),

